vault/logical_raw.go

```diff
@@ -72,23 +82,31 @@ func (b *RawBackend) handleRawRead(ctx context.Context, req *logical.Request, da
 		return nil, nil
 	}
 
-	// Run this through the decompression helper to see if it's been compressed.
-	// If the input contained the compression canary, `outputBytes` will hold
-	// the decompressed data. If the input was not compressed, then `outputBytes`
-	// will be nil.
-	outputBytes, _, err := compressutil.Decompress(entry.Value)
-	if err != nil {
-		return handleErrorNoReadOnlyForward(err)
+	valueBytes := entry.Value
+	if compressed {
+		// Run this through the decompression helper to see if it's been compressed.
+		// If the input contained the compression canary, `valueBytes` will hold
+		// the decompressed data. If the input was not compressed, then `valueBytes`
+		// will be nil.
+		valueBytes, _, err = compressutil.Decompress(entry.Value)
+		if err != nil {
+			return handleErrorNoReadOnlyForward(err)
+		}
+
+		// `valueBytes` is nil if the input is uncompressed. In that case set it to the original input.
+		if valueBytes == nil {
+			valueBytes = entry.Value
+		}
 	}
 
-	// `outputBytes` is nil if the input is uncompressed. In that case set it to the original input.
-	if outputBytes == nil {
-		outputBytes = entry.Value
+	var value interface{} = string(valueBytes)
+	if encoding == "base64" {
```

Comment:
Should we return an error if encoding is anything other than `base64` or empty string?

Reply:
Added a check.